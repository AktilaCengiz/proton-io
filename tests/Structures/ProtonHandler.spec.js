/* eslint-disable no-undef */
/* eslint-disable max-classes-per-file */
const ProtonClient = require("../../src/Structures/ProtonClient");
const ProtonHandler = require("../../src/Structures/ProtonHandler");
const ProtonModule = require("../../src/Structures/ProtonModule");
const checkEqualKeys = require("../checkEqualKeys");
const equal = require("../equal");

const moduleDir = `${process.cwd()}/tests/Structures/Modules`;

class Client extends ProtonClient {
    constructor() {
        super({
            intents: []
        });

        this.handler = new ProtonHandler(this, {
            directory: moduleDir,
            automateCategories: true
        });
    }
}

const client = new Client();

beforeEach(() => client.handler.removeAll() /** Nice work */);

test("load", () => {
    class LocalModule extends ProtonModule {
        constructor() {
            super("A");
        }
    }

    const loaded = client.handler.load(LocalModule);

    checkEqualKeys(loaded, {
        id: "A",
        category: "default",
        filepath: null,
        client,
        handler: client.handler
    });
    const loadedSec = client.handler.load(`${moduleDir}/ExampleModule`);

    checkEqualKeys(loadedSec, {
        id: "ExampleModule",
        category: "Modules",
        client,
        handler: client.handler
    });

    equal(typeof loadedSec.filepath, "string");
    expect(client.handler).toMatchSnapshot();
});

test("loadAll", () => {
    equal(client.handler.modules.size, 0);
    client.handler.loadAll();
    expect(client.handler.modules.size).toBeGreaterThanOrEqual(1);
});

test("remove", () => {
    client.handler.loadAll();
    equal(client.handler.modules.size, 1);
    client.handler.remove(client.handler.modules.first()?.id);
    equal(client.handler.modules.size, 0);
});
