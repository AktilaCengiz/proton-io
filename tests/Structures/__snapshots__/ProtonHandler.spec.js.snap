// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load 1`] = `
ProtonHandler {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "automateCategories": true,
  "client": Object {
    "application": null,
    "channels": Array [],
    "guilds": Array [],
    "owners": Array [],
    "presence": undefined,
    "shard": null,
    "user": null,
    "users": Array [],
  },
  "directory": "/home/heiwa/wioenena/workspace/packages/javascript/node/proton-io/tests/Structures/Modules",
  "modules": Array [
    LocalModule {
      "category": "default",
      "client": Object {
        "application": null,
        "channels": Array [],
        "guilds": Array [],
        "owners": Array [],
        "presence": undefined,
        "shard": null,
        "user": null,
        "users": Array [],
      },
      "filepath": null,
      "handler": [Circular],
      "id": "A",
    },
    Object {
      "category": "Modules",
      "client": Object {
        "application": null,
        "channels": Array [],
        "guilds": Array [],
        "owners": Array [],
        "presence": undefined,
        "shard": null,
        "user": null,
        "users": Array [],
      },
      "filepath": "/home/heiwa/wioenena/workspace/packages/javascript/node/proton-io/tests/Structures/Modules/ExampleModule",
      "handler": [Circular],
      "id": "ExampleModule",
    },
  ],
  Symbol(kCapture): false,
}
`;
